src/shapes/object.class.js

```diff
@@ -1225,13 +1225,13 @@
         this._renderBackground(ctx);
       }
       this._render(ctx);
-      this._drawClipPath(ctx);
+      this._drawClipPath(ctx, this.clipPath);
       this.fill = originalFill;
       this.stroke = originalStroke;
     },
 
-    _drawClipPath: function(ctx) {
-      var path = this.clipPath;
+    _drawClipPath: function (ctx, path) {
+      path = path || this.clipPath;
```

Comment:
since we are going for 5.0 and the _drawClipPath calls aren't many, please make path a required argument and let's skip the  `path = path || this.clipPath;`

Reply:
same for `drawClipPathOnCache`

Reply:
6a73507a